docs/src/modules/components/InterfaceApiPage.js

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable react/no-danger */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { exactProp } from '@mui/utils';
+import Typography from '@mui/material/Typography';
+import Alert from '@mui/material/Alert';
+import VerifiedRoundedIcon from '@mui/icons-material/VerifiedRounded';
+import { alpha } from '@mui/material/styles';
+import { useTranslate, useUserLanguage } from 'docs/src/modules/utils/i18n';
+import HighlightedCode from 'docs/src/modules/components/HighlightedCode';
+import MarkdownElement from 'docs/src/modules/components/MarkdownElement';
+import AppLayoutDocs from 'docs/src/modules/components/AppLayoutDocs';
+import PropertiesSection from 'docs/src/modules/components/ApiPage/sections/PropertiesSection';
+import { DEFAULT_API_LAYOUT_STORAGE_KEYS } from 'docs/src/modules/components/ApiPage/sections/ToggleDisplayOption';
+
+export function getTranslatedHeader(t, header) {
+  const translations = {
+    demos: t('api-docs.demos'),
+    import: t('api-docs.import'),
+  };
+
+  // TODO Drop runtime type-checking once we type-check this file
+  if (!translations.hasOwnProperty(header)) {
+    throw new TypeError(
+      `Unable to translate header '${header}'. Did you mean one of '${Object.keys(
+        translations,
+      ).join("', '")}'`,
+    );
+  }
+
+  return translations[header] || header;
+}
+
+function Heading(props) {
+  const { hash, level: Level = 'h2' } = props;
+  const t = useTranslate();
+
+  return (
+    <Level id={hash}>
+      {getTranslatedHeader(t, hash)}
+      <a aria-labelledby={hash} className="anchor-link" href={`#${hash}`} tabIndex={-1}>
+        <svg>
+          <use xlinkHref="#anchor-link-icon" />
+        </svg>
+      </a>
+    </Level>
+  );
+}
+
+Heading.propTypes = {
+  hash: PropTypes.string.isRequired,
+  level: PropTypes.string,
+};
+
+export default function ApiPage(props) {
+  const {
+    descriptions,
+    pageContent,
+    defaultLayout = 'expanded',
+    layoutStorageKey = DEFAULT_API_LAYOUT_STORAGE_KEYS,
+  } = props;
+  const t = useTranslate();
+  const userLanguage = useUserLanguage();
+
+  const { demos, filename = '', properties } = pageContent;
+
+  const { componentDescription, propertiesDescriptions, interfaceDescription } =
+    descriptions[userLanguage];
+  const description = t('api-docs.pageDescription').replace(/{{name}}/, pageContent.name);
+
+  // Prefer linking the .tsx or .d.ts for the "Edit this page" link.
+  const apiSourceLocation = filename.replace('.js', '.d.ts');
+
+  return (
+    <AppLayoutDocs
+      description={description}
+      disableToc={false}
+      toc={[]}
+      location={apiSourceLocation}
```

Comment:
The "Edit this page" button is broken – it links to https://github.com/mui/mui-x/settings/master
Previously, we were linking to the `.md` file, but did it make much sense if the file was autogenerated?

I think we should remove the "Edit this page" button from the interfaces pages.
What do you think?

Reply:
The link point to `https://github.com/mui/mui-x/edit/master/[the location]` but the location is an empty string. And `https://github.com/mui/mui-x/edit/master/` goes to settings because it's a folder, not a file

Will be fixed with this PR.

https://github.com/mui/material-ui/pull/41702